release.yaml

```diff
@@ -1,13 +1,17 @@
-fleet-agent:
-  - 104.0.1+up0.10.1-rc.1
 fleet:
```

Comment:
The changes in this file are a result of doing `make validate` and `make check-release-yaml`. I have made zero manual changes to this.

Reply:
`charts-build-scripts` is really only checking that everything that is going to be released is included, it does not check that things are should not be released are not included. In that process it might add unexpected things to `release.yaml`

Reply:
IMO doing `make validate` is sorting stuff alphabetically. However, the YAML it's generating is broken. That is fixed by running `make check-release-yaml`. That's been my observation so far.

Reply:
Yeah that whole, `check-release-yaml` scripts stuff is annoying. If I remember correctly the yaml itself isn't "broken" just not in the desired format (whitespce vs no whitespace in array data).

> Well, those changes are based on make check-release-yaml.

`check-release.yaml` does not modify data, it basically just runs it through `yq` and checks if that changed release.yaml. Those changes are coming from `make validate` which does a whole lot more than just alphabetize `release.yaml` (although I can remember if thats done in `check-release-yaml` or `make validate`). `make validate` basically compares the local list of charts with those found in the last release, and reports on anything that should be in released and wasn't. If you wanna see the nitty gritty look here: https://github.com/rancher/charts-build-scripts/blob/master/pkg/validate/validate.go.

We should not be adding charts to `release.yaml` unless we made changes to that chart in the same PR. We've all been guilty of doing this before since they are easy to miss changes, so don't sweat it :)

Reply:
> check-release.yaml does not modify data, it basically just runs it through yq and checks if that changed release.yaml

The `scripts/check-release-yaml` has this `yq -i release.yaml` which modifies `release.yaml` in place. But you're right that it's coming from `make validate` in this case.

> make validate which does a whole lot more than just alphabetize release.yaml (although I can remember if thats done in check-release-yaml or make validate). make validate basically compares the local list of charts with those found in the last release, and reports on anything that should be in released and wasn't. 

Yep, makes sense. In case of this PR, it's alphabetizing as well as adding the rc chart to the `relesae.yaml`.

> We should not be adding charts to release.yaml unless we made changes to that chart in the same PR. We've all been guilty of doing this before since they are easy to miss changes, so don't sweat it :)

I'm lost. Should this PR have the changes to `release.yaml` or not? Anyway, let's discuss this over a call for quicker resolution.